cantools/database/can/signal.py

```diff
@@ -243,21 +256,8 @@ def __init__(self,
         self._is_multiplexer = is_multiplexer
         self._multiplexer_ids = multiplexer_ids
         self._multiplexer_signal = multiplexer_signal
-        self._is_float = is_float
         self._spn = spn
 
-    @property
-    def name(self) -> str:
```

Comment:
the performance improvement is great, but I'd like to keep the "officially supported" attributes somehow documented in the class' docstring. is it somehow possible to add a dummy member in the class for this, something like:

```python
class Message:
  # [...]
  name = "<undefined>"
  """The signal name as a string.
  """
```

(unfortunately the docstring will be ignored, but at least  the `name` attribute will be visible without instantiating an object...)

a slightly more verbose but IMO better alternative would be to keep the property for the docstring and overriding it in the constructor:

```python
class Message:
  def __init__(self, name : str):
    self.name = name

  @property
  def name(self) -> str:
        """The signal name as a string.
        """
        raise RuntimeError("Undefined property") 
```

the main drawback of this is that `help(messageObject.name)` will produce the documentation of the build-in string type instead of the docstring of the property. (`help(messageObject)` and `help(Message.name)` are fine, though.)

Reply:
The attributes with `#:`
```
        #: The signal name as a string.
        self.name: str = name

        #: The scale factor of the signal value.
        self.scale: float = scale

        #: The offset of the signal value.
        self.offset: float = offset
```
will continue to be described in the docs. But `help()` would not work though since the attributes don't have a `__doc__` method.
Alternatively we could document the attributes in the class docstring like [this](https://stackoverflow.com/a/8659919/13170690)

Reply:
> The attributes with `#:`

neat. I didn't know that so far :)

> will continue to be described in the docs. But help() would not work though since the attributes don't have a __doc__ method.

While obviously not 100% optimal, I think that this is an acceptable trade-off for improved performance.

> Alternatively we could document the attributes in the class docstring like [this](https://stackoverflow.com/a/8659919/13170690)

I'd prefer to keep the documentation as close to the code as possible. That's because I am of the opinions that it is next to impossible to keep it complete and consistent otherwise. (besides, putting this into the docstring of the class would still not change the "documentation of object attributes" issue...)